tests/test_streams.py

```diff
@@ -27,3 +28,21 @@ def test_selection(self):
         self.assertEqual([video], container.streams.get(video=(0,)))
 
         # TODO: Find something in the fate suite with video, audio, and subtitles.
+
+    def test_stream_properties(self):
+        # Ensure that all stream properties have sensible values even
+        # if writing to the stream hasn't started yet
+        with av.open(self.sandboxed("output.mp4"), "w") as container:
+            stream = container.add_stream('h264')
```

Comment:
Why is the type unknown when *we* add a stream?

Reply:
It could be. I believe that it is because the codec context is only "opened" on first use.

TBH, I only noticedthis since accessing the `__repr__` when running my code interactively cause things to crash.

Reply:
You are correct, with #1062 (and my "fix" removed) applied I should update this test with:

```
diff --git a/tests/test_streams.py b/tests/test_streams.py
index efee029..496d32d 100644
--- a/tests/test_streams.py
+++ b/tests/test_streams.py
@@ -45,4 +45,4 @@ class TestStreams(TestCase):
             assert stream.duration is None
             assert stream.frames == 0
             assert stream.language is None
-            assert stream.type is None
+            assert stream.type == 'video'
```